posts/2023/05-06-bypassing_tasks_for_test_cases.md

```diff
@@ -0,0 +1,201 @@
+==title==
+Bypassing tasks for test cases
+
+==author==
+Cornelia Kelinske
+
+==description==
+I've been working a lot with external APIs and, as a result, with sandboxing and mocking API responses for test cases.
+These mocks work great unless your production code outsources the API call to a separate task, which means
+another process. In those cases, the mock response set in the test is assigned to the parent process, and the test will throw an error because it 
+cannot find the mock for the outsourced API call. Here is a little workaround I found:
+
+
+==tags==
+coding, elixir, testing
+
+==body==
+
+# 1. What are we testing?
+
+
+Let's say we want to implement an `Oban.Worker` that periodically retrieves a list of thousands of user IDs (e.g. from a .tsv file) and then queries an external API for information
+on these IDs. Our worker has to make one API call per ID. Once all the information is retrieved, the worker will process it and store it in a database or cache.
+Our code might look like this:
+
+```elixir
+use Oban.Worker,
+    max_attempts: 3,
+    queue: :user_info_scraper
+
+  @impl true
+  def perform(_job) do
+    user_ids = SomeOtherModule.get_user_ids()
```

Comment:
Maube cut down the example a bit? You don't need to have an Oban job to explain the Task.async_stream.

Make the example as simple as possible for the reader?

Reply:
Thanks, that is a very good idea!